data/src/main/java/com/example/data/shared_preferences/SharedPreferencesManagerImpl.kt

```diff
@@ -21,13 +25,35 @@ class SharedPreferencesManagerImpl @Inject constructor(
         return Single.just(token)
     }
 
+    override fun getAdditionalUserInfo(): Single<AdditionalUserInfo> {
+        val data = AdditionalUserInfo(
+            login = preferences.getString(LOGIN, EMPTY_STRING) ?: EMPTY_STRING,
+            imageUri = preferences.getString(IMAGE, EMPTY_STRING)?.toUri() ?: EMPTY_STRING.toUri()
+        )
+        return Single.just(data)
+    }
+
     override fun writeToken(token: String) {
         val editor = preferences.edit()
```

Comment:
let's create private methods here something like `saveValue(key, value)`
In those methods, you will do the magic like 
`editor
            .putString(TOKEN, token)
            .apply()` and save other data
F.E. your `writeUserLogin(...)` will looks like
`override fun writeUserLogin(login: String) {
        saveString(LOGIN, login)
    }`

Reply:
done